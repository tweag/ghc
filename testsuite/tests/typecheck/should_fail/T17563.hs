{-# LANGUAGE QuantifiedConstraints #-}

module T17563 where

blah :: (forall a. a b ~ a c) => b -> c
blah = undefined
