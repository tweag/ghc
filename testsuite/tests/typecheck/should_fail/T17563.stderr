
T17563.hs:5:9: error:
    • Illegal equational constraint a b ~ a c
      (Use GADTs or TypeFamilies to permit this)
    • In the quantified constraint ‘forall (a :: * -> *). a b ~ a c’
      In the type signature: blah :: (forall a. a b ~ a c) => b -> c
